### PR TITLE
fix(Wallet): fix amount checking, compare to maxAmount

### PR DESCRIPTION
**Pull request by UncleBill**
closes #

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const TransferERC20 = memo<TransferERC20Props>(({ token }) => {
     )
 
     // transfer amount
-    const transferAmount = new BigNumber(amount || '0').multipliedBy(pow10(token.decimals)).toFixed()
+    const transferAmount = formatAmount(amount, token.decimals)
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:43]: The format amount is an approximation amount. We need it to be a big int in WEI.

**Comment by guanbinrui** [comment]
@UncleBill, close this PR?

File: packages/web3-shared/src/utils/formatter.ts [hunk 2]
```diff
@@ -14,8 +14,8 @@ export function formatPrice(price: BigNumber.Value, decimalPlaces = 6) {
     return new BigNumber(price).decimalPlaces(decimalPlaces).toString()
 }
 
-export function formatAmount(amount: BigNumber.Value = '0', decimals = 0) {
-    return new BigNumber(amount).multipliedBy(pow10(decimals)).toFixed()
+export function formatAmount(amount: BigNumber.Value, decimals = 0) {
+    return new BigNumber(amount || '0').multipliedBy(pow10(decimals)).toFixed()
```
Comment (septs) [inline packages/web3-shared/src/utils/formatter.ts:18]: why?

Comment (UncleBill) [reply packages/web3-shared/src/utils/formatter.ts:18]: If `amount` from user input is empty string `''`, here will result `NaN`.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**